force-app/main/default/classes/CaseTriggerHelperTest.cls

```diff
@@ -0,0 +1,103 @@
+
+        List<Account> hospitalDepartmentList = TestDataFactory.createHospitalsDepartmentList(3, hospitalList, false);
+        hospitalDepartmentList[0].Department_type__c = 'Children';
+        hospitalDepartmentList[0].Department__c = 'Surgery';
+        hospitalDepartmentList[1].Department_type__c = 'Adult';
+        hospitalDepartmentList[1].Department__c = 'Psychiatric';
+        hospitalDepartmentList[2].Department_type__c = 'Adult';
+        hospitalDepartmentList[2].Department__c = 'Therapy';
+        insert hospitalDepartmentList;
+
+        List<Patient__c> patientList = TestDataFactory.createPatientList(2, hospitalList, false);
+        patientList[0].Birthday_date__c = Date.valueOf('2008-9-25');
+        patientList[0].First_Name__c = 'Young';
+        patientList[1].Birthday_date__c = Date.valueOf('2000-10-13');
+        patientList[1].First_Name__c = 'Old';
+        insert patientList;
+
+        List<InsuranceCompany__c> insuranceCompanyList = TestDataFactory.createInsuranceCompanyList(1, true);
+
+        List<Insurance__c> insuranceAdultList = TestDataFactory.createInsuranceList(1, 299, insuranceCompanyList, new List<Patient__c>{patientList[1]}, false);
+        insuranceAdultList[0].Covered_departments__c = 'Reception;Therapy;Gastroenterology;Surgery;Psychiatric';
+        insuranceAdultList[0].Department_type__c = 'Adult';
+        insert insuranceAdultList;
+
+        List<Insurance__c> insuranceChildrenList = TestDataFactory.createInsuranceList(1, 299, insuranceCompanyList, new List<Patient__c>{patientList[0]}, false);
+        insuranceChildrenList[0].Covered_departments__c = 'Reception;Therapy;Gastroenterology;Surgery';
+        insuranceChildrenList[0].Department_type__c = 'Children';
+        insert insuranceChildrenList;
+        
+        List<Case> treatmentList = new List<Case>();
+        for (Integer i = 0; i < 4; i++){
+            treatmentList.add(new Case());
+        }
+        treatmentList[0].Patient__c = patientList[0].Id;
+        treatmentList[0].Insurance__c= insuranceChildrenList[0].Id;
+        treatmentList[0].AccountId = hospitalDepartmentList[0].Id;
+        treatmentList[1].Patient__c = patientList[0].Id;
+        treatmentList[1].Insurance__c= insuranceChildrenList[0].Id;
+        treatmentList[1].AccountId = hospitalDepartmentList[1].Id;
+        treatmentList[2].Patient__c = patientList[1].Id;
+        treatmentList[2].Insurance__c= insuranceAdultList[0].Id;
+        treatmentList[2].AccountId = hospitalDepartmentList[0].Id;
+        treatmentList[3].Patient__c = patientList[1].Id;
+        treatmentList[3].Insurance__c= insuranceAdultList[0].Id;
+        treatmentList[3].AccountId = hospitalDepartmentList[2].Id;
+        
+        Test.startTest();
+        List<Database.SaveResult> insertResults = Database.insert (treatmentList, false);
+        Test.stopTest();
+        
+        System.assertEquals(true,insertResults[0].isSuccess());
```

Comment:
Please use Assert class and use method with 3 params(the third one is error message when values are not the same)